src/main/java/facades/SearchFacade_Impl.java

```diff
@@ -245,13 +248,9 @@ public PersonDTO_OUT deletePerson(Integer id) {
             em.getTransaction().begin();
             // Find Person from ID
             Person person = em.find(Person.class, id);
-            if (person == null) {
+            if (person == null || person.getAddress() != null) {
                 throw new WebApplicationException("Could not delete Person. Provided ID does not exist.", 400);
             }
-            // Remove all the phones
-            person.getPhones().forEach((phone) -> {
-                em.remove(phone);
-            });
```

Comment:
Maybe if we remove Person first, we can do this afterwards? Haven't tried, though.